ddl/column.go

```diff
@@ -301,10 +299,33 @@ func onAddColumns(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error
 		}
 	})
 
-	tblInfo, columnInfos, positions, offsets, err := checkAddAndCreateColumnInfos(t, job)
+	tblInfo, columnInfos, columns, positions, offsets, ifNotExists, err := checkAddColumns(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
+	if len(columnInfos) == 0 {
+		if len(columns) == 0 {
```

Comment:
If the length of the columns is 0, it can't be put into job queues. It will filter in [line2438](https://github.com/pingcap/tidb/pull/15540/files#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2R2438).